File: STRIVER/video/vid-5/pattern18/p18.cpp
// E
// D E
// C D E
// B C D E
// A B C D E 

#include <iostream>
using namespace std;

int main()
{
    int row = 5;

    for (char i = 0; i < row; i++){

        char ch = (row-i-1)+'A';
        for(int j = 0; j<= i; j++){
            cout<<ch<<" ";
            ch++;
        }
        cout<<endl;
    }

    return 0;
}

File: STRIVER/video/vid-5/pattern18/p18.py
# E
# D E
# C D E
# B C D E
# A B C D E 

def p18(row:int)->None:

    for i in range(row+1):
        ch = (row-i-1)+ord('A')+1
        for j in range(i):
            print(f"{chr(ch)} ",end="")
            ch+=1
        print(" ")

row = 5
p18(row)
